File: lab4 mpp/TeledonDomain/Donatie.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace teledon.domain
{
    public class Donatie : Entity<long>
    {
        public Donator Don { get; set; }
        
        public CazCaritabil Caz { get; set; }
        public int Suma { get; set; }

        public Donatie(CazCaritabil Caz, Donator Don, int Suma)
        {
            this.Don = Don;
            this.Suma = Suma;
            this.Caz = Caz;
        }
    }
}

File: GUImpp/TeledonModel/src/main/java/teledon/model/Donatie.java
package teledon.model;

public class Donatie extends Entity<Long> {

    public Donator donator;
    public CazCaritabil caz;
    public int suma;

    @Override
    public String toString() {
        return "teledon.model.Donatie{" +
                "idDonatie="  +
                ", donator=" + donator +
                ", caz=" + caz +
                ", suma=" + suma +
                '}';
    }

    public Donatie(Donator donator, CazCaritabil caz, int suma) {
        this.donator = donator;
        this.suma = suma;
        this.caz = caz;
    }
    public Donatie() {}
    public CazCaritabil getCaz() {
        return this.caz;
    }

    public Donator getDonator() {
        return donator;
    }

    public void setDonator(Donator donator) {
        this.donator = donator;
    }

    public void setCaz(CazCaritabil c) {
        this.caz = c;
    }

    public int getSuma() {
        return suma;
    }

    public void setSuma(int suma) {
        this.suma = suma;
    }
}
